### PR TITLE
Allow to customize our babel configuration

### PR DESCRIPTION
**Pull request by arunoda**
This works for both for all of the app pages including the custom document component.

### CODE REVIEW

**Comment by rauchg** [comment]
do expressions 😍

File: README.md [hunk 1]
```diff
@@ -396,10 +396,40 @@ In order to extend our usage of `webpack`, you can define a function that extend
...
+
 module.exports = {
-  webpack: (cfg, { dev }) => {
-    cfg.module.rules.push({ test: /\.svg$/, loader: 'babel!react-svg' })
-    return cfg
+  // config is the set of options we pass to our babel-loaders's query option
+  babel: function (config, { dev }) {
+    // Add the stage-0 preset.
+    // Make sure to use 'require.resolve' otherwise we won't be able to find it.
+    config.presets.push(require.resolve('babel-preset-stage-0'))
```
Comment (nkzawa) [inline README.md]: This seems a bit too hacky to recommend. I wonder if the following works. 

```js
import presetStage0 from 'babel-preset-stage0'

...

config.presets.push(presetStage0)
```

Comment (arunoda) [inline README.md]: We can't import since it's next.config.js. It didn't work for me for simply giving the preset. 

I will try more options.

**Comment by nkzawa** [comment]
Another idea: just use `.babelrc` if it exists. I can’t remember if we already discussed this option.
Then you can just set `presets: ['next/babel']` to extend the default setting.

File: README.md (hunk 1 above)
Comment (arunoda) [inline README.md]: That's a pretty good option. I like it. 
I thought we wanted to expose everything via next.config.js. 

@rauchg what do you think?

**Comment by arunoda** [comment]
That's a pretty good option. I like it.
I thought we wanted to expose everything via next.config.js.

@rauchg what do you think?

**Comment by siddharthkp** [comment]
@nkzawa, that sounds 😍
Is it safe to assume next/babel would always be applicable and apply it anyways?

**Pull request merged**